src/user-profile.js

```diff
@@ -196,16 +202,24 @@ const Profile = {
       if (fs.existsSync(oldAddonsDir) &&
           fs.lstatSync(oldAddonsDir).isDirectory()) {
         const fnames = fs.readdirSync(oldAddonsDir);
+        const addonRenames = [];
         for (const fname of fnames) {
           const oldFname = path.join(oldAddonsDir, fname);
           const newFname = path.join(this.addonsDir, fname);
           const lstat = fs.lstatSync(oldFname);
 
           if (fname !== 'plugin' && lstat.isDirectory()) {
             // Move existing add-ons.
-            this.renameDir(oldFname, newFname);
+            addonRenames.push(this.renameDir(oldFname, newFname));
           }
         }
+
+        Promise.all(addonRenames).then(() => {
```

Comment:
Couldn't there still be a race condition if AddonManager is in the middle of loading things when this hits?